source/modulo_new_core/src/translators/WriteStateConversion.cpp

```diff
@@ -1,30 +1,65 @@
-#include "modulo_core/communication/message_passing/WriteStateConversion.hpp"
+#include "modulo_new_core/translators/WriteStateConversion.hpp"
 
-#include <state_representation/exceptions/IncompatibleReferenceFramesException.hpp>
+#include <std_msgs/msg/bool.hpp>
+#include <std_msgs/msg/float64.hpp>
+#include <std_msgs/msg/float64_multi_array.hpp>
+#include <std_msgs/msg/string.hpp>
 
 using namespace state_representation::exceptions;
 
-namespace modulo::core::communication::state_conversion {
+namespace modulo_new_core::translators {
+
+static void write_point(geometry_msgs::msg::Point& msg, const Eigen::Vector3d& vector) {
+  msg.x = vector.x();
+  msg.y = vector.y();
+  msg.z = vector.z();
+}
+
+static void write_vector3(geometry_msgs::msg::Vector3& msg, const Eigen::Vector3d& vector) {
+  msg.x = vector.x();
+  msg.y = vector.y();
+  msg.z = vector.z();
+}
+
+static void write_quaternion(geometry_msgs::msg::Quaternion& msg, const Eigen::Quaterniond& quat) {
+  msg.w = quat.w();
+  msg.x = quat.x();
+  msg.y = quat.y();
+  msg.z = quat.z();
+}
+
+void write_msg(geometry_msgs::msg::Point& msg, const state_representation::CartesianState& state, const rclcpp::Time&) {
+  if (state.is_empty()) throw EmptyStateException(state.get_name() + " state is empty while attempting to publish it");
```

Comment:
The inline `if` statements in this file should also be split with newlines and braces